contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1503,6 +1505,43 @@ pre_transform_target_entry(ResTarget *res, ParseState *pstate,
 				alias_len = strlen(identifier_name);
 				colname_start = pstate->p_sourcetext + res->location;
 			}
+			if(list_length(cref->fields) > 1 &&
+                IsA(llast(cref->fields), String))
+            {
+                identifier_name = strVal(llast(cref->fields));
+                alias_len = strlen(identifier_name);
+                colname_start = pstate->p_sourcetext + res->location;
+                while(true)
+                {	/*To check how many open sqb are present in sourcetext. */
+                    if(*colname_start == '[')
+                    {
+                        open_square_bracket++;
+                    }
+                    if(*colname_start == ']')
+                    {
+                        open_square_bracket--;
+                    }
+					/*
+					 * last_dot pointer is to trace the last dot in the sourcetext,
+					 * as last dot indicates the starting of column name.
+					 */
+                    if(open_square_bracket == 0 && *colname_start == '.')
+                    {
+                        last_dot = colname_start;
+                    }
+					/* 
+					 * If there is no open sqb and colname_start is at ' ' , 
+					 * It means, colname_start is at the end of column name.
+					 */
+                    if(open_square_bracket == 0 && (*colname_start == ' ' || *colname_start == ','))
+                    {
+						last_dot++;
+                        colname_start = last_dot;
+                        break;
+                    }
+					colname_start++;
+                }
+            }
```

Comment:
Nit: Indentation seems a bit off for the whole block.
Additionally, the approach taken here seem to have a couple of flaws:
1. We are assuming that the identifier can only be quoted with brackets `[]` but that's not true. `"` can also be used for quoting.
2. We are looking for `' '` and `,` to find the end of identifier but don't forget that the identifier itself can contain both of these special characters. For example a column name can be: `[ABC, DEF]` (contains both comma and space)
Overall, we need to be very careful in checks and conditions here or otherwise the result might be completely wrong.

Reply:
I agree with RIshabh, can you check implementation of SplitGUCList to see if we can do some similar implementation for our use case?

Reply:
1. We are checking, if number of open brackets are zero or not. If there is no bracket and we reach at either space or comma, it means we are at the end of the column name.
2. `create table t ("ABc", int)`, this type of syntax is not supported. We can only pass column name either with square brackets or with no brackets
3. I will fix the indentation in next commit.

Reply:
> I agree with RIshabh, can you check implementation of SplitGUCList to see if we can do some similar implementation for our use case?


I will check this.

Reply:
Resolved.